cli/src/semgrep/state.py

```diff
@@ -30,6 +32,7 @@ class SemgrepState:
     """
 
     app_session: AppSession = Factory(AppSession)
+    request_id: UUID = Factory(uuid4)
```

Comment:
Can we avoid yet another global ...? Is there no other way?

Reply:
I don't think there are 'easy' solutions to this problem without leveraging global state. 

The use case here requires that whenever something in `semgrep ci` fails this uuid gets sent to the fail-open endpoint so we can accurately correlate the backend requests with the failures and I don't think it's reasonable to rewrite the entire error handling implementation to solve this metrics gap.